modules/llm-cache/ds/kv_state_cache_manager.cc

```diff
@@ -25,21 +25,32 @@ limitations under the License.
 
 namespace vineyard {
 
-KVStateCacheManager::KVStateCacheManager(int dimension, int cacheCapacity,
-                                         int layer, int blockSize,
-                                         int syncInterval, std::string socket) {
+KVStateCacheManager::KVStateCacheManager(
+    Client& client, std::shared_ptr<KVStateCacheBuilder>& cache,
+    int syncInterval, std::string& llmCacheSyncLock,
+    std::string& llmCacheObjectName)
+    : client(client) {
   this->syncInterval = syncInterval;
-  VLOG(100) << "socket:" << socket;
-  client.Connect(socket);
+  this->kvStateCacheBuilder = cache;
+  this->llmCacheSyncLock = llmCacheSyncLock;
+  this->llmCacheObjectName = llmCacheObjectName;
+  this->syncThread = new std::thread(SyncThreadFunc, this);
```

Comment:
No `std::thread *` please. Why a pointer is needed here?

Reply:
OK